halo2_proofs/src/plonk.rs

```diff
@@ -159,6 +229,43 @@ impl<C: CurveAffine> ProvingKey<C> {
     pub fn get_vk(&self) -> &VerifyingKey<C> {
         &self.vk
     }
+    /// Writes a proving key to a buffer.
+    /// Does so by first writing the verifying key and then serializing the rest of the data (in the form of field polynomials)
+    pub fn write<W: io::Write>(&self, writer: &mut W) -> Result<(), Error> {
+        self.vk.write(writer)?;
+        let partial_pkey = ProvingKeyWithoutVerifyingKey {
+            l0: self.l0.clone(),
+            l_last: self.l_last.clone(),
+            l_active_row: self.l_active_row.clone(),
+            fixed_values: self.fixed_values.clone(),
+            fixed_polys: self.fixed_polys.clone(),
+            fixed_cosets: self.fixed_cosets.clone(),
+            permutation: self.permutation.clone(),
+        };
+        bincode::serialize_into(writer, &partial_pkey).map_err(Error::Serde)
```

Comment:
Hardcoding `bincode` directly here is a bit tough TBH. But I understand that you can't use serde traits because that could allow yaml or anything esentially.

As said in my review comment, I'm a bit hessitant over the `serde`.

Reply:
Previously because of evaluation there was too much to implement `write` for directly, but since Han pointed out that we can recompute evaluation directly, it seems the only primitive necessary is to implement write for `Polynomial<F, _> = Vec<F>` and `Vec<Polynomial<F, _>>`, which seems doable. 

So if you want to limit dependencies, I think it is possible to just implement some `write` and `read` directly.

Reply:
I'd prefer it and avoid serde. Unless it is extremely painful.
If any of Han or Onur agree with me, I'd suggest to change it.